src/main/java/com/cleanroommc/groovyscript/compat/mods/prodigytech/AtomicReshaper.java

```diff
@@ -0,0 +1,166 @@
+package com.cleanroommc.groovyscript.compat.mods.prodigytech;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.helper.ingredient.OreDictIngredient;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import lykrast.prodigytech.common.recipe.AtomicReshaperManager;
+import lykrast.prodigytech.common.util.Config;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RegistryDescription
+public class AtomicReshaper extends VirtualizedRegistry<AtomicReshaperManager.AtomicReshaperRecipe> {
+    @RecipeBuilderDescription(example = {
+        @Example(".input(item('minecraft:gold_ingot')).output(item('minecraft:emerald_block')).primordium(10).time(50)"),
+        @Example(".input(item('minecraft:gold_block')).addOutput(item('minecraft:diamond_block'), 10).addOutput(item('minecraft:carrot'), 3).primordium(7)")
+    })
+    public AtomicReshaper.RecipeBuilder recipeBuilder() {
+        return new AtomicReshaper.RecipeBuilder();
+    }
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(recipe -> {
+            if (recipe.isOreRecipe())
+                AtomicReshaperManager.INSTANCE.removeOreRecipe(recipe.getOreInput());
+            else
+                AtomicReshaperManager.INSTANCE.removeRecipe(recipe.getInput());
+        });
+        restoreFromBackup().forEach(AtomicReshaperManager.INSTANCE::addRecipe);
+    }
+
+    @MethodDescription(example = @Example("ore('paper')"))
+    public boolean removeByInput(IIngredient input) {
+        if (input instanceof OreDictIngredient) {
+            AtomicReshaperManager.AtomicReshaperRecipe recipe = AtomicReshaperManager.INSTANCE.removeOreRecipe(((OreDictIngredient) input).getOreDict());
+            if (recipe == null)
+                return false;
+            addBackup(recipe);
+            return true;
+        } else {
+            boolean success = false;
+            for (ItemStack it : input.getMatchingStacks()) {
+                AtomicReshaperManager.AtomicReshaperRecipe recipe = AtomicReshaperManager.INSTANCE.removeRecipe(it);
+                if (recipe != null) {
+                    success = true;
+                    addBackup(recipe);
+                }
+            }
+            return success;
+        }
+    }
+
+    public void add(AtomicReshaperManager.AtomicReshaperRecipe recipe) {
+        AtomicReshaperManager.INSTANCE.addRecipe(recipe);
+        addScripted(recipe);
+    }
+
+    @MethodDescription(priority = 2000, example = @Example(commented = true))
+    public void removeAll() {
+        // NOTE: can't be rolled back due to ItemMap<T> being protected

```

Comment:
in situations like this, you should create an accessor to access the field, such as we do for Thermal Expansion: https://github.com/CleanroomMC/GroovyScript/blob/master/src/main/java/com/cleanroommc/groovyscript/core/mixin/thermalexpansion/BrewerManagerAccessor.java

it takes a few steps, as you need to do three things:
- add the modid to `LateMixin`
- create file called `mixin.groovyscript.[modid].json` in `resources`
- create the accessor and add it to the above json

Reply:
It turns out I didn't actually have to do this and went through all this for nothing 😭 oh well